### PR TITLE
Travis CI: Add Python 3.9 to the testing

### PR DESCRIPTION
**Pull request by cclauss**
Removes the stages to allow all four jobs can run in parallel.

~Scipy is not yet compatible with Python 3.9 so run that job in `allow_failures` mode.~

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,33 +18,36 @@ env:
...
-    - stage: "test python 3.7"
-      name: "Run tests on python 3.7"
-      script: pytest
-      python: 3.7
-    - stage: "test python 3.8"
-      name: "Run tests on python 3.8"
-      script: pytest
-      python: 3.8
+before_script:
+  - sleep 15 # mongo takes time to start
```
Comment (marload) [inline .travis.yml:47]: In my experience, it often takes more than 15 seconds to install mongodb. It seems better to increase this time to 30seconds.

Comment (cclauss) [reply .travis.yml:47]: mongodb should be kicked off on line 19 and then we pip install a lot of stuff and that takes more than a full minute...

Look at the times along the right margin of the Travis output...
Screenshot 2020-10-27 at 14 51 20

**Comment by marctorsoc** [comment]
> Removes the stages to allow all four jobs can run in parallel.
> 
> Scipy is not yet compatible with Python 3.9 so run that job in `allow_failures` mode.

I'm not sure about supporting python 3.9 if scipy is not compatible with it. Isn't better to wait until that happens rather than the allow_failures hack?

**Comment by marctorsoc** [comment]
> > Removes the stages to allow all four jobs can run in parallel.
> > Scipy is not yet compatible with Python 3.9 so run that job in `allow_failures` mode.
> 
> I'm not sure about supporting python 3.9 if scipy is not compatible with it. Isn't better to wait until that happens rather than the allow_failures hack?

@cclauss ?

**Comment by cclauss** [comment]
Scipy now has Python 3.9 wheels and these tests pass on Py39.

**Review by marctorsoc — APPROVED** [review]
LGTM, thanks!

**Pull request merged**